public/app/features/dashboard-scene/scene/ShareQueryDataProvider.ts

```diff
@@ -24,27 +25,69 @@ export class ShareQueryDataProvider extends SceneObjectBase<ShareQueryDataProvid
   private _results = new ReplaySubject<SceneDataProviderResult>();
   private _sourceProvider?: SceneDataProvider;
   private _passContainerWidth = false;
+  private _shouldUnwireClonedData = false;
 
   constructor(state: ShareQueryDataProviderState) {
-    super(state);
-
-    this.addActivationHandler(() => {
-      // TODO handle changes to query model (changed panelId / withTransforms)
-      //this.subscribeToState(this._onStateChanged);
+    super({
+      data: emptyPanelData,
+      ...state,
+    });
 
-      this._subscribeToSource();
+    this.addActivationHandler(this._onActivate);
+  }
 
-      return () => {
-        if (this._querySub) {
-          this._querySub.unsubscribe();
-        }
-        if (this._sourceDataDeactivationHandler) {
-          this._sourceDataDeactivationHandler();
-        }
-      };
-    });
+  // Will detect the root of the scene and if it's a PanelEditor, it will clone the data from the source panel.
+  // We are doing it because the source panel's original scene (dashboard) does not exist in the edit mode.
+  private _setupEditMode(panelId: number, root: PanelEditor) {
+    this._shouldUnwireClonedData = false;
+
+    const keyToFind = getVizPanelKeyForPanelId(panelId);
+    const source = findObjectInScene(
+      root.state.dashboardRef.resolve(),
+      (scene: SceneObject) => scene.state.key === keyToFind
+    );
+
+    if (source) {
+      // Indicate that when de-activated, i.e. navigating back to dashboard, the cloned$data state needs to be removed
+      // so that the panel on the dashboar can use data from the actual source panel.
+      this._shouldUnwireClonedData = true;
+      this.setState({
+        $data: source.state.$data!.clone(),
+      });
+    }
   }
 
+  private _onActivate = () => {
+    const root = this.getRoot();
+    if (root instanceof PanelEditor && this.state.query.panelId) {
+      this._setupEditMode(this.state.query.panelId, root);
+    }
+    // TODO handle changes to query model (changed panelId / withTransforms)
+    this.subscribeToState(this._onStateChanged);
+    /**
+     * If the panel uses a shared query, we clone the source runner and attach it as a data provider for the shared one.
+     * This way the source panel does not to be present in the edit scene hierarchy.
+     */
+
+    this._subscribeToSource();
+
+    return () => {
+      if (this._querySub) {
+        this._querySub.unsubscribe();
+      }
+      if (this._sourceDataDeactivationHandler) {
+        this._sourceDataDeactivationHandler();
+        this._sourceDataDeactivationHandler = undefined;
+      }
+
+      if (this._shouldUnwireClonedData) {
+        this.setState({
+          $data: undefined,
+        });
+      }
```

Comment:
is this really needed? seems to do the same as https://github.com/grafana/grafana/pull/79991/files#diff-a2c47e2e69d74212db99d6b8b23eaea0cf85c28ec2184a4e01573287ca451198R113 

it's strange to have both :)

Reply:
My bad - forgot to remove this. It's not relevant for the `ShareQueryDataProvider`, it's onle important in the edit mode. Good catch, thank you